File: fizzbuzz/fizzbuzz.js
#!/usr/bin/env node

fs = require('fs');

function multiple(x, n) {
    return x % n == 0;
}

function fizzbuzz_str(a, b, i) {
    if (multiple(i, a)) {
        if (multiple(i, b)) {
            return "FB";
        } else {
            return "F";
        }
    } else if (multiple(i, b)) {
        return "B";
    } else {
        return i;
    }
}

function fizzbuzz(a, b, n) {
    var result = '';

    for (var i = 1; i <= n; i++) {
        result += fizzbuzz_str(a, b, i);
        if (i < n) result += ' ';
    }

    console.log(result);
}

fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line != "") {       
        line = line.replace(/\r/gm, ""); // For Windows.

        var arr = line.split(" "); // Splits the string by space
        fizzbuzz(arr[0], arr[1], arr[2]);
    }
})

File: fizzbuzz/fizzbuzz.pl
#!/usr/bin/env perl

sub fizzbuzz {
    my ( $a, $b, $n ) = @_;

    for ( 1 .. $n ) {
        my $fizz = $_ % $a == 0;
        my $buzz = $_ % $b == 0;

        print "FB" if ($fizz && $buzz);
        print "F" if ($fizz && !$buzz);
        print "B" if (!$fizz && $buzz);
        print $_ if (!$fizz && !$buzz);
        print " " if $_ < $n;
    }

    print "\n";
}

open FILE, "<", $ARGV[0] or die $!;
map { fizzbuzz(split / /) } <FILE>;
